tools/wavetool.py

```diff
@@ -0,0 +1,146 @@
+        right_idx = right_idx - 1
+    return wave[left_idx:right_idx,:], left_idx
+
+# float point binary comparison is not supported, and will not be supported
+# check recorded wave through smart amplifier component
+def compare_wav_smart_amp(wave, ref_wave, fs):
+    trimed_ref_wave, _ = trim_wave(ref_wave)
+    # compare the first two channel
+    trimed_wave_ch_0_1, delay1 = trim_wave(wave[:,0:2])
+    trimed_ref_wave = trimed_ref_wave[0:trimed_wave_ch_0_1.shape[0],:]
+    compare_result0_1 = np.array_equal(trimed_ref_wave, trimed_wave_ch_0_1)
+    # compare the second two channel
+    trimed_wave_ch_2_3, delay2 = trim_wave(wave[:,2:4])
+    trimed_ref_wave = trimed_ref_wave[0:trimed_wave_ch_2_3.shape[0],:]
+    compare_result2_3 = np.array_equal(trimed_ref_wave, trimed_wave_ch_2_3)
+
+    smart_amp_delay = ((delay2 - delay1) / fs * 1000)
+    print("Delay of smart amplifier is %0.3fms" % smart_amp_delay)
+    if compare_result0_1 and compare_result2_3 and smart_amp_delay < 5.:
+        print('Recorded wave is binary same as reference wave')
+        print("Wave comparison result: PASSED")
+    else:
+        print('Recorded wave is not binary same as reference wave')
+        print("Wave comparison result: FAILED")
+        sys.exit(1001)
+
+def parse_cmdline():
+    parser = argparse.ArgumentParser(add_help=True, formatter_class=argparse.RawTextHelpFormatter,
+        description='A Tool to Generate and Manipulate Wave Files.')
+    parser.add_argument('-v', '--version', action='version', version='%(prog)s 1.0')
+    # wave parameters
+    parser.add_argument('-g', '--generate', type=str, choices=['sine','cosine'], help='generate specified types of wave')
+    parser.add_argument('-A', '--amp', type=float, nargs='+', default=[1.0], help='amplitude of generated wave')
+    parser.add_argument('-F', '--freq', type=float, nargs='+', default=[997.0], help='frequency of generated wave')
+    parser.add_argument('-P', '--phase', type=float, nargs='+', default=[0.0], help='phase of generated wave')
+    parser.add_argument('-D', '--duration', type=float, nargs='+', default=[10.], help='duration of generated wave')
+    parser.add_argument('-S', '--sample_rate', type=int, default=48000, help='sample rate of generated wave')
+    parser.add_argument('-C', '--channel', type=int, default=2, help='channels of generated wave')
+    parser.add_argument('-B', '--bits', type=str, choices=['S8', 'S16', 'S32', 'F32'], default='S16',
+    help='sample bits of generated wave')
+    parser.add_argument('-o', '--output', type=str, help='path to store generated files', default='.')
+    # wave comparison arguments
+    parser.add_argument('-c', '--compare', type=str, choices=['smart_amp'], help='compare recorded wave and reference wave')
+    parser.add_argument('-R', '--recorded_wave', type=str, help='path of recorded wave')
+    parser.add_argument('-r', '--reference_wave', type=str, help='path of reference wave')
+    return vars(parser.parse_args())
```

Comment:
What's wrong with `cmd_args.generate`, `cmd_args.compare`, etc.?

Reply:
didn't totally get the point here, sorry.

Reply:
What is the purpose of `vars()`? Why not `return parse.parse_args()`?